xonsh/tools.py

```diff
@@ -1593,26 +1614,50 @@ def expand_case_matching(s):
     return ''.join(t)
 
 
-def globpath(s, ignore_case=False, return_empty=False):
+def globpath(s, ignore_case=False, return_empty=False, sort_result=None):
     """Simple wrapper around glob that also expands home and env vars."""
-    o, s = _iglobpath(s, ignore_case=ignore_case)
+    if sort_result is None:
+        sort_result = builtins.__xonsh_env__.get('GLOB_SORTED')
+
+    o, s = _iglobpath(s, ignore_case=ignore_case, sort_result=sort_result)
     o = list(o)
     no_match = [] if return_empty else [s]
     return o if len(o) != 0 else no_match
 
 
-def _iglobpath(s, ignore_case=False):
+def _iglobpath(s, sort_result, ignore_case=False):
     s = builtins.__xonsh_expand_path__(s)
+
     if ignore_case:
         s = expand_case_matching(s)
+
     if sys.version_info > (3, 5):
         if '**' in s and '**/*' not in s:
             s = s.replace('**', '**/*')
+
         # `recursive` is only a 3.5+ kwarg.
-        return glob.iglob(s, recursive=True), s
+        if sort_result:
+            paths = glob.glob(s, recursive=True)
+            paths.sort()
+            paths = iter(paths)
+        else:
+            paths = glob.iglob(s, recursive=True)
+
+        return paths, s
     else:
-        return glob.iglob(s), s
+        if sort_result:
+            paths = glob.glob(s)
+            paths.sort()
+            paths = iter(paths)
+        else:
+            paths = glob.iglob(s)
```

Comment:
Yes, it does, the github window is unfortunate:

        if sort_result:
            paths = glob.glob(s, recursive=True)
            paths.sort()
            paths = iter(paths)
        else:
            paths = glob.iglob(s, recursive=True)
        return paths, s